src/local/h2_adjust/units.py

```diff
@@ -1,15 +1,115 @@
 """
 Additional project specific units
 """
-from scmdata.units import UNIT_REGISTRY  # type: ignore
+from typing import Any
+
+import scmdata
+from scmdata.units import UNIT_REGISTRY, UnitConverter
+
+from local.h2_adjust.exceptions import SanitizeError
 
 if not hasattr(UNIT_REGISTRY, "hydrogen"):
     UNIT_REGISTRY.define("H = [hydrogen] = H2")
     UNIT_REGISTRY.define("hydrogen = H")
     UNIT_REGISTRY.define("t{symbol} = t * {symbol}".format(symbol="H"))
 
-    UNIT_REGISTRY.define("4/16 * H = CH4")
-    UNIT_REGISTRY.define(f"{14 + 2 * 16}H = NOx")
-    UNIT_REGISTRY.define("3/17 H = NH3")
+__all__ = [
+    "UNIT_REGISTRY",
+    "sanitize_combustion_intensity_units",
+    "sanitize_production_intensity_units",
+]
+
+
+def _get_unit_scaling(source_unit: str | Any, target_unit: str) -> float:
+    if not isinstance(source_unit, str):
+        return 1.0
+    else:
+        uc = UnitConverter(source_unit, target_unit)
+
+        return uc.convert_from(1)
+
+
+def sanitize_combustion_intensity_units(
+    intensities: scmdata.ScmRun, energy_unit: str = "MWh", mass_unit: str = "kg"
+) -> scmdata.ScmRun:
+    """
+    Convert emissions intensities to a common set of units
+
+    Default target unit is "kg X / MWh"
+    """
+
+    def _convert_intensities(ts: scmdata.ScmRun):
+        product: str = ts.get_unique_meta("product", True)
+        unit: str = ts.get_unique_meta("unit", True)
+        target_unit = f"{mass_unit} {product} / {energy_unit}"
+
+        scale = _get_unit_scaling(unit, target_unit)
+        result = ts * scale
+        result["unit"] = target_unit
+        return result
+
+    return intensities.apply(_convert_intensities)
+
+
+def h2_mass_factor(unit: str, species: str):
+    """
+    H2 -> species == kg species / kg H2
+    """
+    if species == "H2":
+        factor = 1
+    elif species == "CH4":
+        factor = (12 + 4) / 4
+    elif species == "NH3":
+        factor = (14 + 3) / 3
+    elif species == "NOx":
+        # Assumes mass equivalence and NO2
+        factor = 14 + 2 * 16
```

Comment:
Missing factor of 2? I'm not really sure what the mass equivalence we're assuming here is or means

Reply:
We are assuming that 1 kg of H2 = 46 kg NOx.

In this case, yes I think we do need a factor of 2 if we are just assuming atomic mass equivalence

Reply:
Yep, see above. I think the confusion is I don't know what reaction we are assuming here in order to get the conversion...

Reply:
Ha turns out it isn't used anymore - let's remove as I cannot defend the assumptions made

Reply:
Nice